File: cinebook-backend/cinebook-backend/src/main/java/com/cinebook/cinebook/service/DashBoardService.java
package com.cinebook.cinebook.service;

import com.cinebook.cinebook.dto.response.*;
import org.hibernate.validator.constraints.Length;
import org.hibernate.validator.constraints.pl.REGON;

import java.util.List;

public interface DashBoardService {
      DashBoardResponseDto getDashBoard();

      //find the total booking of the per movie
      MovieBookingStatsDto getMostBookedMovie();
      //find the total revenue according to the movie
      MovieRevenueDto getHighestRevenueMovie();

      //find the status report of the booking  by the status of booking
      List<BookingStatusResponseDto>getBookingTotalByStatus();
      //find the status report of the booking by payment status
      List<PaymentStatusResponseDto>getBookingTotalByPaymentStatus();

}

File: cinebook-frontend/src/services/dashboardService.js
import axiosInstance from "./axiosinstance";

export const getDashBoard = () => {
    return axiosInstance.get("/api/dashboard");
};

export const getMostBookedMovie = () => {
    return axiosInstance.get("/api/dashboard/most-booked-movie");
};

export const getHighestRevenueMovie = () => {
    return axiosInstance.get("/api/dashboard/highest-revenue-movie");
};

export const getBookingStatusReport = () => {
    return axiosInstance.get("/api/dashboard/booking-status-report");
};

export const getPaymentStatusReport = () => {
    return axiosInstance.get(
        "/api/dashboard/booking-Payment-status-report"
    );
};